### ISSUE TITLE
Help with optimization

### ISSUE THREAD

**Issue by RomkeVdMeulen** [aurelia/documentation#291]
I would love to see a one page summary, like the cheat sheet, of some of the basic and more advanced techniques to optimize Aurelia performance. Or a section, in the style of [Securing Your App](http://aurelia.io/hub.html#/doc/article/aurelia/framework/latest/securing-your-app/1).

There's a lot to learn in the documentation already, but it's spread out, and doesn't cover all common scenarios. There's more information on StackOverflow and in blogposts, but sometimes it's hard to judge how up-to-date and well-accepted such information is.

For example, I've been looking into what the best way is to expose computed information from my ViewModel. There's the [Binding: Computed Properties](http://aurelia.io/hub.html#/doc/article/aurelia/binding/latest/binding-computed-properties/1) page, but that only shows some simple examples. It doesn't show that `@computedFrom` can [handle paths](https://github.com/aurelia/binding/issues/149), or how to handle observation of arrays. I would also love to have a better grasp on when to use `@computedFrom`, when to use other techniques, and when dirty-checking is acceptable (though I understand this may be a controversial topic for discussion in the docs).

Then there's `ObserverLocator`. I've read blogposts recommending I have it injected into my ViewModels and use it to write my own observer code. But the API doc describes this class as "internal". Should I be using `BindingEngine.propertyObserver` and `BindingEngine.collectionObserver` in stead?

[Delegate vs. Trigger](http://aurelia.io/hub.html#/doc/article/aurelia/binding/latest/binding-delegate-vs-trigger) is also very helpful, and could be linked to.

And should I be worrying about such details beforehand? Or should I profile my app once I've got it working to look for bottlenecks? Searching for "Profiling" or "Profiler" on the Hub shows nothing. Do I profile it like any JS application? Or are there specific things to look for when using Aurelia?

In short, a page in the doc to give me an overview of considerations, with links to more in-depth information and discussion, would be a great help for more advanced applications.

**Comment by EisenbergEffect** [comment]
@jdanyow May be one of the best people to do this. I'm not sure what his work load is right now. I think it's a great idea though.

**Comment by arkadiuszizycki** [comment]
What is a status of that issue?
